api/grpc/mpi/v1/command.proto

```diff
@@ -0,0 +1,358 @@
+// These proto definitions follow https://protobuf.dev/programming-guides/style/
+// and recommendations outlined in https://static.sched.com/hosted_files/kccncna17/ad/2017%20CloudNativeCon%20-%20Mod%20gRPC%20Services.pdf
+syntax = "proto3";
+package f5.nginx.agent.api.grpc.mpi.v1;
+
+option go_package = "github.com/nginx/agent/v3/api/grpc/mpi/v1";
+
+import "common.proto";
+import "file.proto";
+import "google/protobuf/empty.proto";
+import "google/protobuf/struct.proto";
+import "google/protobuf/timestamp.proto";
+
+// A service outlining the command and control options for a DataPlane
+service CommandService {
+    // Connects NGINX Agents to the Management Plane agnostic of instance data
+    rpc Connect(ConnectionRequest) returns (ConnectionResponse) {}
+    // Reports on instances and their configurations
+    rpc Status(DataPlaneStatus) returns (google.protobuf.Empty) {}
+    // Reports on instance health
+    rpc Health(DataPlaneHealth) returns (google.protobuf.Empty) {}
+    // A decoupled communication mechanism between the data plane and management plane.
+    rpc Subscribe(stream DataPlaneMessage) returns (stream ManagementPlaneMessage) {}
+}
+
+// The connection request is an intial handshake to establish a connection, sending NGINX Agent instance information
+message ConnectionRequest {
+    // Meta-information associated with a message
+    f5.nginx.agent.api.grpc.mpi.v1.common.MessageRequest message_metadata = 1;
+    // instance information associated with the NGINX Agent
+    Instance agent = 2;
+}
+
+// This contains a series of NGINX Agent configurations
+message AgentConfig { 
+    // Command server settings
+    Command command = 1;
+    // Metrics server settings
+    Metrics metrics = 2;
+    // A series of key/value pairs to add more data to the NGINX Agent instance
+    repeated google.protobuf.Struct labels = 3;
```

Comment:
`google.protobuf.Struct` is just `map<string, [Value]>` as per https://protobuf.dev/reference/protobuf/google.protobuf/#struct, where `Value` can be different things. https://protobuf.dev/reference/protobuf/google.protobuf/#value. Should we just go with `map<string, string>` ?

Reply:
The values can be lists so I think this type is best

Reply:
probably require discussion

Reply:
I'd rather keep the structure the way it is. If it is desired not to consume the full flexibility of the structures offered, that's fine, but other consumers of this API might want to